runners/spark/src/main/java/org/apache/beam/runners/spark/TestSparkRunner.java

```diff
@@ -78,6 +102,33 @@ public SparkPipelineResult run(Pipeline pipeline) {
     result.waitUntilFinish();
     assertThat(result, testPipelineOptions.getOnCreateMatcher());
     assertThat(result, testPipelineOptions.getOnSuccessMatcher());
+    // if the pipeline was executed in streaming mode, validate aggregators.
+    if (isStreaming) {
+      // validate assertion succeeded (at least once).
+      int success = result.getAggregatorValue(PAssert.SUCCESS_COUNTER, Integer.class);
```

Comment:
For now, I don't see a reason to extend this.
I'm pretty sure `FAILURE_COUNTER` will never show as spark Accumulators don't report back if task failed (thrown Exception), but I'm asserting this anyway.
The `SUCCESS_COUNTER` assertion is important just to make sure any assertion occurred - this is because in case no data is processed (for whatever reason) tests will succeeded without actually asserting anything.
So basically I'm asserting that _any_ assertion happened.
If needed, this could be extended in the future. For now, I'll merge.